api/provider/aws/task_read.go

```diff
@@ -1,9 +1,90 @@
 package aws
 
 import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
 	"github.com/quintilesims/layer0/common/models"
 )
 
 func (t *TaskProvider) Read(taskID string) (*models.Task, error) {
-	return nil, nil
+	// todo: use tlake's
+	environmentID, err := t.lookupTaskEnvironmentID(taskID)
+	if err != nil {
+		return nil, err
+	}
+	fqEnvironmentID := addLayer0Prefix(t.Config.Instance(), environmentID)
+
+	taskARN, err := t.lookupTaskARN(taskID)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterName := fqEnvironmentID
+	if _, err := t.readTask(clusterName, taskARN); err != nil {
+		return nil, err
+	}
+
+	model := &models.Task{
+		TaskID:        taskID,
+		EnvironmentID: environmentID,
+	}
+
+	return model, nil
+}
+
+// todo remove
+func (t *TaskProvider) lookupTaskEnvironmentID(taskID string) (string, error) {
+	tags, err := t.TagStore.SelectByTypeAndID("task", taskID)
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("environment_id").First(); ok {
+		return tag.Value, nil
+	}
+
+	return "", fmt.Errorf("Failed to find environment_id for task '%s'", taskID)
+}
+
+func (t *TaskProvider) readTask(clusterName, taskARN string) (*ecs.Task, error) {
+	input := &ecs.DescribeTasksInput{}
+	input.SetCluster(clusterName)
+	input.SetTasks([]*string{aws.String(taskARN)})
+
+	if err := input.Validate(); err != nil {
+		return nil, err
+	}
+
+	// todo: catch does ntoe xist error
```

Comment:
When you work on this, unfortunately DescribeTasks() also does not have a notion of Task not found: http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTasks.html#API_DescribeTasks_Errors. The closest equivalent you can use from this is `ClusterNotFoundException`